### ISSUE TITLE
Crashes after loading a game (Achievements related)

### ISSUE THREAD

**Issue by awsms** [libretro/RetroArch#10978]
## Description

While playing The Sims 2 (USA, Europe) with the achivements enabled in the settings, RetroArch will try to unlock a bunch of achievements (I never played this game before), then instantly crash after the game is loaded. If the achivements aren't enabled, the emulation works just fine.

### Expected behavior

Running the game normally

### Actual behavior

A bunch of achievements are triggered, then retroarch crashes after a single frame is loaded.

### Steps to reproduce the bug

1. Load any GBA core
2. Enable the achievements in the settings
3. Run The Sims 2, Europe

### Version/Commit
You can find this information under Information/System Information

Retroarch 1.8.9 (Git 2e115ba) / From the Ubuntu repos

### Environment information

- OS: Linux Mint 19.3
- Compiler: -

I tried all of the GBA emulators and I get the same exact bug on all, so I'm pretty sure this isn't core related. Hash of the game tested: 51455002b0daa27de8eaa5d9ad1afd6f

[retrobug_sims_vba_next.log](https://github.com/libretro/RetroArch/files/4880252/retrobug_sims_vba_next.log)
[retrobug_sims_mgba.log](https://github.com/libretro/RetroArch/files/4880253/retrobug_sims_mgba.log)

**Comment by Jamiras** [comment]
I can confirm that the three achievements shown in your log file unlock when loading this game with the BIOS startup screen enabled. However, I did not experience the crash. I expect you've encountered the issue fixed by #10971. On other games where the achievements don't immediately unlock, do you still crash when an achievement triggers? Is there any way you can try the latest nightly to see if the issue persists?

**Comment by awsms** [comment]
I'll try the new build in a sec. I don't remember having encountered any crash related to the achievements unlocking though, except this one.

However, I noticed a lot of random crashes at the startup of a game, but there wasn't any relevent infos in the logs, this line being the last one before the crash:
```[INFO] [RCHEEVOS]: posted playing activity```
I encountered it in a lot of different games, regardless of the platform (neogeo, snes, n64 etc). The game can be loaded fine 5 times, then it will crash the 6th.
